src/colorizer/utils/data_load_utils.ts

```diff
@@ -96,7 +96,10 @@ export async function loadFromParquetUrl<T extends FeatureDataType>(url: string,
  * @param zipFile A ZIP file object.
  * @returns A Promise that resolves to a map of file paths to a File object.
  */
-export async function zipToFileMap(zipFile: File): Promise<Record<string, File>> {
+export async function zipToFileMap(
+  zipFile: File,
+  onLoadProgress?: ReportLoadProgressCallback
```

Comment:
Added support for reporting load progress, similar to how it's done in `Dataset`.